src/varname.jl

```diff
@@ -26,10 +29,10 @@ julia> @varname x[:, 1][1+1]
 x[:,1][2]
 ```
 """
-struct VarName{sym, T<:Tuple}
+struct VarName{sym, T}
     indexing::T
 
-    VarName{sym}(indexing::Tuple=()) where {sym} = new{sym,typeof(indexing)}(indexing)
+    VarName{sym}(indexing=()) where {sym} = new{sym,typeof(indexing)}(indexing)
```

Comment:
We could entirely get rid of tuples and use `IdentityLens` instead.

(And then require `T<:Lens` or so.)

Reply:
Not opposed to doing that. I was only slightly worried that that this would have some consequences downstream and was thinking that we could support both. Buuuut it's probably a good idea to just tear that bandaid off immediately :shrug: